appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/utils/MockCollectorClient.java

```diff
@@ -153,8 +153,7 @@ public List<ResourceScopeMetric> getMetrics(Set<String> presentMetrics) {
                       .map(x -> x.getName())
                       .collect(Collectors.toSet());
 
-              return (!exported.isEmpty() && current.size() == exported.size())
-                  && receivedMetrics.containsAll(presentMetrics);
+              return (!exported.isEmpty() && receivedMetrics.containsAll(presentMetrics));
```

Comment:
Removing the size check will prevent us from capturing metrics that are unexpectedly generated.